File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.go
func largestAltitude(gain []int) int {
    height, highest := 0, 0
    for _, g := range gain {
        height += g
        highest = max(highest, height)
    }
    return highest
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.c
int largestAltitude(int* gain, int gain_size){
    int h = 0, hmax = 0;
    for (int i = 0; i < gain_size; i++) {
        h += gain[i];
        hmax = (hmax >= h) ? hmax : h;
    }
    return hmax;
}